src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the component of the Duke program
+ * that deals with interactions with the user.
+ */
+public class Ui {
+
+    protected static String lineSpacing = "____________________________________________________________";
```

Comment:
Perhaps consider making this as a constant that can be used with the final keyword and changing the variable name to "LINE_SPACING".